src/vs/platform/request/node/proxy.ts

```diff
@@ -44,7 +47,8 @@ export async function getProxyAgent(rawRequestURL: string, env: typeof process.e
 		rejectUnauthorized: isBoolean(options.strictSSL) ? options.strictSSL : true,
 	};
 
+	// {{SQL CARBON EDIT}} Using updated 'http-proxy-agent' and 'https-proxy-agent' packages.
 	return requestURL.protocol === 'http:'
-		? new (await import('http-proxy-agent'))(opts as any as Url)
-		: new (await import('https-proxy-agent'))(opts);
+		? new HttpProxyAgent(opts)
```

Comment:
It would be best to keep the logic of doing the import inline here so we keep the current behavior of not loading the package unless it's actually needed (a proxy URL is specified). It may not matter, but unless we know for sure that always loading it isn't going to cause problems it's safer to try and reduce behavior changes as much as we can.

Reply:
I don't think that should be a concern.

Reply:
Why risk it? Unless we know for sure that it won't have any side effects I'm not sure we should be changing behavior. Sometimes libraries can do things at a global scope that may impact other functionality. (generally they shouldn't of course but it's hard to know for sure)

Reply:
Is there another way to do that? 
Because I get below error when using inline import as above, that's the reason I had to import the type specifically.
`Only a void function can be called with the 'new' keyword.ts(2350)`